### PR TITLE
Rework 'make generate-images'

### LINKED ISSUE #12314: `make generate-images` fails with newer inkscape versions
- Gitea version (or commit ref): 7744b36
- Git version: not relevant
- Operating system: not relevant
- Database (use `[x]`): not relevant
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

`make generate-images` fails with newer inkscape versions (version 1.0) because some CLI arguments have changed.

References: 
https://inkscape.org/sk/doc/inkscape-man.html
https://wiki.inkscape.org/wiki/index.php/Using_the_Command_Line#Changes_from_0.92

Instead of `inkscape -f <infile> ... -e <outfile>`, it should be `inkscape <infile> ... --export-file=<outfile>`.

(I stumbled upon this when creating my own Gitea logo)

This could be helpful:
`inkscape --version 2>/dev/null | cut -d ' ' -f 2 | cut -d . -f 1` prints the major inkscape version which then can be compared to '0' or '1'.

### PR DESCRIPTION
**Pull request by silverwind**
- Remove external dependencies and replace it with a node script that does does the same.
- Move detail removal from gitea-sm.png to favicon.png
- Remove favicon.ico and its generation, it is unused and we already serve favicon.png in its place.

Fixes: https://github.com/go-gitea/gitea/issues/12314

### CODE REVIEW

**Comment by stertingen** [comment]
Thanks for this PR as this will fix my initial problem. (https://github.com/go-gitea/gitea/issues/12314)

But isn't it a bit overkill to download chromium just to build a bunch of png images? Why not use the original SVG image instead of the PNG images?

logo.svg: 5.6K (base image size)
gitea-lg.png: 33K
gitea-512.png: 17K
gitea-192.png: 4.2K
gitea-sm.png: 3.2K
favicon.png: 4.1K
apple-touch-icon.png: 4.1K

In some cases, the resulting PNG files are much bigger than the original SVG file, even after compression. I see that favicon and apple-touch-icon have to be a rasterized, but using the SVG for the larger variants might reduce the amount and size of transmitted images.

**Comment by silverwind** [comment]
> But isn't it a bit overkill to download chromium just to build a bunch of png images

Probably is, but `svgexport` was just the first cross-platform tool I've found that was up to the job that can be easily installed with only our existing dependencies on Node/Go and which does not require messing with `PATH` (like Inkscape would).

> Why not use the original SVG image instead of the PNG images?

I agree we should ideally only use the SVG image in the UI but it'd be quite a bit of work and I think some parts of the code can only work with bitmap images (avatar) and then there's the issue that not every image can be represented as an SVG (think photos).

**Comment by silverwind** [comment]
Updated to use a node script and two npm modules in place of previous Chromium-based solution. It uses JSDOM's canvas renderer under the hood whose SVG implementation is probably not as complete as Chrome's but seems to work fine for our logo.

Those modules are now only installed on demand, otherwise we would gain 215 new transitive `devDependencies` which I think are better avoided for such a rarely used task.

Reviewers should check the diff on `public/img/gitea-sm.png` which gained some detail. I think the change is fine to keep.

**Comment by stertingen** [comment]
I like this better, thanks!

**Comment by lafriks** [comment]
You could just use domxml to remove layer in svg before generating gitea-sm

**Comment by silverwind** [comment]
https://github.com/xmldom/xmldom you mean? Yeah, certainly possible and probably better than the string modification.

I'm still contemplating whether to add the dependencies to `devDependencies`. Probably better in the long run.

**Comment by silverwind** [comment]
I think I will actually do the layer removal but only for the favicon because that is rendered the smallest (16px):

image

`gitea-sm` is used on the page header at 30px and I think it would benefit from the detail, especially on hidpi screens:

image

Regarding `devDependencies`: I will keep it the current way because zopflipng requires a native dependency and that's a no-go for our bundled `node_modules` in the src tarballs.

**Comment by lafriks** [comment]
Yes, xmldom :) It's already dependency for svg2img so that would not add much to node_modules size

**Comment by silverwind** [comment]
Done. Detail is now removed from `favicon.png` instead of `gitea-sm.png` previously.

File: build/generate-images.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  if (bg) {
+    const rect = parser.parseFromString('<rect width="100%" height="100%" fill="white"/>');
+    document.documentElement.insertBefore(rect, document.documentElement.firstChild);
+  }
+
+  if (removeDetail) {
+    for (const el of Array.from(document.getElementsByTagName('g') || [])) {
+      for (const attribute of Array.from(el.attributes || [])) {
+        if (attribute.name === 'class' && attribute.value === 'detail-remove') {
```
Comment (lafriks) [inline build/generate-images.js:31]: Why not just use id attribute and layer3 as value?

Comment (silverwind) [reply build/generate-images.js:31]: I added that class specifically so users can run their own SVGs through it and `layer3` is specific to our source image so could destroy parts of the user image.

**Comment by silverwind** [comment]
Updated to use `fabric` to render into the canvas which allows us to skip a few wonky dependencies.

File: build/generate-images.js (hunk 1 above)
Comment (silverwind) [reply build/generate-images.js:31]: Also, added this info about `detail-remove` to the docs.

**Review by lafriks — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**